app/lib/screens/study/tasks/task_screen.dart

```diff
@@ -34,23 +34,22 @@ class _TaskScreenState extends State<TaskScreen> {
     taskInstance = TaskInstance.fromInstanceId(widget.taskInstance.id, study: subject!.study);
   }
 
-  Widget? _buildTask() {
-    switch (taskInstance.task.runtimeType) {
-      case CheckmarkTask _:
+  Widget _buildTask() {
+    switch (taskInstance.task) {
+      case CheckmarkTask checkmarkTask:
         return CheckmarkTaskWidget(
-          task: taskInstance.task as CheckmarkTask,
+          task: checkmarkTask,
           key: UniqueKey(),
           completionPeriod: taskInstance.completionPeriod,
         );
-      case QuestionnaireTask _:
+      case QuestionnaireTask questionnaireTask:
         return QuestionnaireTaskWidget(
-          task: taskInstance.task as QuestionnaireTask,
+          task: questionnaireTask,
           key: UniqueKey(),
           completionPeriod: taskInstance.completionPeriod,
         );
       default:
-        print('${taskInstance.task.runtimeType} is not a supported Task!');
-        return null;
+        throw ArgumentError('Task ${taskInstance.task.type} not supported');
```

Comment:
This default switch branch can be prevented in the future if we use sealed classes: https://dart.dev/language/class-modifiers#sealed

Reply:
https://github.com/hpi-studyu/studyu/issues/566